.github/workflows/test_action.yml

```diff
@@ -0,0 +1,49 @@
+on:
+  pull_request:
+    types:
+      - opened
+      - synchronize
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          ref: ${{ github.head_ref }}
```

Comment:
Can you try `base_ref` here? thanks!

Reply:
Well it wouldnt test my code i think because the push would happen to head i think

Reply:
```yaml
ref: ${{github.event.pull_request.head.ref}}
repository: ${{github.event.pull_request.head.repo.full_name}}
```

found this from a gitub discussion (posted by staff)